resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/internal/resources/WorkspacePreferencesTest.java

```diff
@@ -16,13 +16,16 @@
 
 import java.util.*;
 import junit.framework.ComparisonFailure;
+import junit.framework.Test;
 import org.eclipse.core.internal.resources.Workspace;
 import org.eclipse.core.internal.resources.WorkspacePreferences;
 import org.eclipse.core.resources.*;
 import org.eclipse.core.runtime.*;
-import org.eclipse.core.tests.resources.ResourceTest;
+import org.eclipse.core.tests.resources.AutomatedResourceTests;
+import org.eclipse.core.tests.resources.WorkspaceSessionTest;
+import org.eclipse.core.tests.session.WorkspaceSessionTestSuite;
 
-public class WorkspacePreferencesTest extends ResourceTest {
+public class WorkspacePreferencesTest extends WorkspaceSessionTest {
```

Comment:
This should be a `WorkspaceSessionTest` to have the test run with a dedicated workspace setup. Otherwise, the import/export test for preferences leads to a complete reinitialization of the preferences, which miss some of the previously attached listeners. Subsequent tests can fail due to those missing listeners. Having a dedicated workspace setup seems to be the option for test isolation from my point of view.